src/snowflake/snowpark/modin/plugin/_internal/resample_utils.py

```diff
@@ -108,15 +109,15 @@
 )
 
 
-def rule_to_snowflake_width_and_slice_unit(rule: str) -> tuple[int, str]:
+def rule_to_snowflake_width_and_slice_unit(rule: Frequency) -> tuple[int, str]:
```

Comment:
```suggestion
def rule_to_snowflake_width_and_slice_unit(rule: Frequency | str) -> tuple[int, str]:
```
`validate_resample_supported_by_snowflake` still use it and give a string as input.

Reply:
This is not needed. `Frequency` is defined as `Union[str, "BaseOffset"]`